contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/performance_mode_controller.py

```diff
@@ -0,0 +1,196 @@
+    # in ce-oem provider.
+    # It's the empty value by defaul, you have to define it as Checkbox's
+    # configuraion.
+    PERFORMANCE_PYTHON_MODULE_PATH = os.environ.get(
+        "PERFORMANCE_PYTHON_MODULE_PATH", ""
+    )
+    if not os.path.exists(PERFORMANCE_PYTHON_MODULE_PATH):
+        raise FileNotFoundError(
+            (
+                "Fail to get the full path of performance python module. "
+                "Please define the path in Checkbox configuration. "
+                "e.g. PERFORMANCE_PYTHON_MODULE_PATH=/path/of/the/module.py"
+            )
+        )
+    # Get the name of context manager function which controls the performance
+    # mode. Default name is performance_mode. You can assign the specific name
+    # via environment variable
+    PERFORMANCE_FUNCTION_NAME = os.environ.get(
+        "PERFORMANCE_FUNCTION_NAME", "performance_mode"
+    )
+
+    return get_function_from_a_module(
+        module_name=os.path.basename(PERFORMANCE_PYTHON_MODULE_PATH),
+        module_path=PERFORMANCE_PYTHON_MODULE_PATH,
+        function_name=PERFORMANCE_FUNCTION_NAME,
+    )
+
+
+class ModuleLoader:
+    """
+    A class to load Python modules dynamically from a given file path.
+
+    Attributes:
+    -----------
+    module_name : str
+        The name to assign to the module once it's loaded.
+    full_path : str
+        The full file path to the module.
+    module : Optional[object]
+        The loaded module object, initialized to None.
+
+    Methods:
+    --------
+    load_module() -> None:
+        Loads the module from the specified file path.
+    get_function(func_name: str) -> Callable:
+        Retrieves a function from the loaded module.
+    """
+
+    def __init__(self, module_name: str, full_path: str):
+        """
+        Initializes the ModuleLoader with the given module name and path.
+
+        Parameters:
+        -----------
+        module_name : str
+            The name to assign to the module once it's loaded.
+        full_path : str
+            The full file path to the module.
+        """
+        self.module_name = module_name
+        self.full_path = full_path
+        self.module: Optional[object] = None
+
+    def load_module(self) -> None:
+        """
+        Loads the module from the specified file path.
+
+        Raises:
+        -------
+        FileNotFoundError:
+            If the module file does not exist.
+        """
+        try:
+            loader = SourceFileLoader(self.module_name, self.full_path)
+            spec = importlib.util.spec_from_loader(self.module_name, loader)
+            self.module = importlib.util.module_from_spec(spec)
+            loader.exec_module(self.module)
+        except FileNotFoundError:
+            logging.error(
+                "Module '{}' cannot be loaded since '{}' doesn't exist".format(
+                    self.module_name, self.full_path
+                )
```

Comment:
```suggestion
                "Module '%s' cannot be loaded since '%s' doesn't exist",
                self.module_name,
                self.full_path,
```

Reply:
Update